server/container/util/src/main/java/org/apache/james/util/FluentFutureStream.java

```diff
@@ -81,8 +92,8 @@ private FluentFutureStream(CompletableFuture<Stream<T>> completableFuture) {
      * For all values of the underlying stream, an action will be performed.
      */
     public FluentFutureStream<T> performOnAll(Function<T, CompletableFuture<Void>> action) {
-        return FluentFutureStream.of(
-            CompletableFutureUtil.performOnAll(completableFuture(), action));
+        return map(t -> action.apply(t).thenApply(any -> t),
```

Comment:
ideally I think this should be 
``` java
public FluentFutureStream<T> forEach(Consumer<? super T> action) {
        futurStream
            .thenAccept(l->l.stream().forEach(action));
        return this;
 }
```
if you really want to wait until the side effects are done before chaining something along the lines of 
```java
public FluentFutureStream<T> forEach(Consumer<? super T> action) {
        return new FluentFutureStream(
            futurStream
            .thenAcceptAsync(l->l.stream().forEach(action))
            .<ArrayList<T>>thenComposeAsync(ignored-> futurStream)
        );
    }
```
would probably be better.